src/components/transport_manager/src/transport_adapter/transport_adapter_impl.cc

```diff
@@ -774,23 +774,36 @@ void TransportAdapterImpl::RemoveFinalizedConnection(
     const DeviceUID& device_handle, const ApplicationHandle& app_handle) {
   const DeviceUID device_uid = device_handle;
   LOG4CXX_AUTO_TRACE(logger_);
-  sync_primitives::AutoWriteLock lock(connections_lock_);
-  ConnectionMap::iterator it_conn =
-      connections_.find(std::make_pair(device_uid, app_handle));
-  if (it_conn == connections_.end()) {
-    LOG4CXX_WARN(logger_,
-                 "Device_id: " << &device_uid << ", app_handle: " << &app_handle
-                               << " connection not found");
-    return;
+  {
+    sync_primitives::AutoWriteLock lock(connections_lock_);
+    ConnectionMap::iterator it_conn =
+        connections_.find(std::make_pair(device_uid, app_handle));
+    if (it_conn == connections_.end()) {
```

Comment:
@Ypostolov swap if condition

Reply:
@LuxoftAKutsan 
fixed in 4be18be33fc5f36bfe297a0622dcf8c4853a45bc